src/Auth/AwsIamAuth.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace GraphQL\Auth;
+
+use Aws\Credentials\CredentialProvider;
+use GuzzleHttp\Psr7\Request;
+
+class AwsIamAuth implements AuthInterface
+{
+    protected const SERVICE_NAME = 'appsync';
+
+    /**
+     * @param Request $request
+     * @param array $headers
+     * @return Request
+     */
+    public function run(Request $request, array $headers = []): Request
+    {
+        $region = getenv('AWS_REGION');
```

Comment:
The package should not assume that the AWS region is declared in the environment variables, this would be bad design. I would suggest passing the AWS region to the constructor of this class as an alternative.